src/main/java/com/segment/analytics/android/integrations/appboy/AppboyIntegration.java

```diff
@@ -119,16 +128,30 @@ public Appboy getUnderlyingInstance() {
   public void identify(IdentifyPayload identify) {
     super.identify(identify);
 
+
     String userId = identify.userId();
     if (!StringUtils.isNullOrBlank(userId)) {
       mAppboy.changeUser(mUserIdMapper.transformUserId(userId));
+
+      if (mTraitsCache != null) {
+        mTraitsCache.clear();
```

Comment:
Agree with @ggm. @ivngrz we did this together but it was raised in a meeting that this should only happen when the userId is different to the previous one or it was previously null. Including the `changeUser` line, as Braze documentation almost begs to execute the `changeUser` only exceptionally, as it looks like a heavy task for the SDK (I guess it will do things like re-download all user campaigns)

Reply:
@ggm It should be fixed now, could you please check again?